src/http/buf.rs

```diff
@@ -226,6 +227,20 @@ impl MemSlice {
     fn get(&self) -> &[u8] {
         unsafe { &(*self.buf.get())[self.start..self.end] }
     }
+
+    pub fn as_str(&self) -> &str {
```

Comment:
It's because we want to keep the `unsafe` near the proof. The proof that a certain `MemSlice` is valid utf8 is when we look at the `req.path` from httparse. Consider:

```rust
impl Uri {
    pub fn with_some_slice(slice: MemSlice) {
        // save the slice internally
    }

    fn something_else(&self) -> &str {
        unsafe { from_utf8_unchecked(self.mem_slice.as_ref()) }
    }
}
```

Now, what happens if future us (or some other future person), perhaps not remembering that this unsafety exists, creates a `Uri::with_some_slice(my_maybe_not_utf8_slice)` somewhere else in the code. It wasn't unsafe to create the `Uri`, so we don't realize that we did anything unsafe.

Compare:

```rust
impl Uri {
    pub fn with_some_mem_str(str: MemStr) -> Uri {

    }

    fn something_else(&self) -> &str {
        &self.mem_str.as_ref() // not unsafe!
    }
}
```

A future persons then says "I want to create a `Uri` over here, using my `MemSlice`, how do I do that? Oh, it takes a `MemStr`, how do I make that? Uh oh, only with `unsafe from_utf8_unchecked()`, I guess I need to check my `MemSlice` before trying that.